lib/versioning/regex/index.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  compare,
+  eq,
+  gt,
+  ltr,
+  maxSatisfying,
+  minSatisfying,
+  Range,
+  satisfies,
+} from 'semver';
+import { logger } from '../../logger';
+import { RangeStrategy, VersioningApi } from '../common';
+
+const safe = require('safe-regex');
+
+interface Version {
+  // major, minor, and patch act in the standard semver fashion, but without
+  // correctness requirements: if any one or two is omitted, they are treated
+  // as zero. At least one must be present
+  major: number;
+  minor: number;
+  patch: number;
+  // prereleases are treated in the standard semver manner, if present
+  prerelease: string;
+  // architectures, if present, are treated as a compatibility layer: we will
+  // never try to update to a version with a different architecture. Other than
+  // for juding compatibility (exact string match), the architecture is ignored
+  architecture: string;
+}
+
+// config is expected to be overridden by a user-specified RegExp value
+// sample values:
+//
+// * emulates the "semver" configuration:
+//   RegExp('^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(-(?<prerelease>.*))?$')
+// * emulates the "docker" configuration:
+//   RegExp('^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(-(?<architecture>.*))?$')
+// * matches the versoining scheme used by the Python images on DockerHub:
+//   RegExp('^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(?<prerelease>[^.-]+)?(-(?<architecture>.*))?$');
+let config = null;
```

Comment:
```ts
let config: RegExp = null;
```

Reply:
@ViceIce btw have you seen this new feature?

![image](https://user-images.githubusercontent.com/6311784/62996587-80991080-be65-11e9-8946-2428db73040e.png)

Whenever the suggestion is just a straight code change, it's then one click to apply

Reply:
e.g.

\`\`\`suggestion
let config: RegExp = null;
\`\`\`

Reply:
And then the result:
![image](https://user-images.githubusercontent.com/6311784/62996630-a9b9a100-be65-11e9-90e8-ac33a2f2b866.png)

Reply:
Nice, will try it